docs/configuration.md

```diff
@@ -0,0 +1,212 @@
+|-------------|------------------|-------------------------------------------------------------------|
+| name        | String           | Key in the values.yaml file this will be mapped to                |
+| description | String           | Description that will be displayed to the user in the UI on Terra |
+| required    | String           | 'true' or 'false' if this field is required                       |
+| type        | String           | Options: string, shared-volume, exclusive-volume                  |
+
+
+## Field Types
+
+- **string**: A simple text input field
+- **shared-volume**: A shared volume input field, allowing multiple plugins to access the same storage
+- **exclusive-volume**: An exclusive volume input field, ensuring only this plugin can access the storage
+
+### String
+
+Pretend you have a simple Plugin that just needs to receive a string value from the user. Let's say you need to know the 
+uid of the user that should own the application on disk.
+
+1. Define the field in your `terra.yaml`:
+
+    ```yaml linenums="1" title="my-plugin/terra.yaml"
+    fields:
+      - name: user_uid
+        description: "The UID of the user that should own the application on disk"
+        required: true
+        type: string
+    ```
+
+2. In your `values.yaml`, you can set a default value or leave it empty:
+
+    ```yaml linenums="1" title="my-plugin/values.yaml"
+    user_uid: ""
+    ```
+   
+    !!! note "Default Value and IDE Autocompletion"
+        This is actually not needed for Terra to function, but it is good practice to have a 
+        default value or an empty string to avoid errors. It also allows many IDEs to auto 
+        complete for the values in the templates directory.
+
+3. In your chart's template, you can access this value using Helm's templating syntax:
+
+    ```yaml linenums="1" title="my-plugin/templates/configmap.yaml"
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: my-config
+    data:
+      user_uid: {{ .Values.user_uid | quote }}
+    ```
+
+For most cases, the String type can satisfy your needs. Yaml in general will evaluate even integers are strings, so you 
+can use it for any string value you need to pass to your templates.
+
+!!! note "Boolean Values"
+    We do plan on adding a boolean type in the future, but for now, you can use strings like "true" or "false"
+
+### Shared Volume
+
+Shared Volumes are flagged by Genesis to be mounted by all workstations launched in the target project. This means
+Terra can be used to install software to shared volumes that can be run by multiple workstations. This is useful for
+example when you want to install something like an AppImage that can be run by any workstation in the project. This
+means you can keep the software installed only once and all workstations can continue to be small in size.
+
+Storage volumes are managed upstream by the Genesis Storage system. This means you can request the user select
```

Comment:
link to the storage docs if available.

Reply:
These docs aren't available yet. We can make a note for when they are ready.